Lib/fontbakery/specifications/googlefonts.py

```diff
@@ -3750,6 +3751,44 @@ def com_google_fonts_check_integer_ppem_if_hinted(ttFont):
                  " will be rounded into and integer value.")
 
 
+@condition
+def gfonts_repo_structure(font):
+  """ The family at the given font path
+      follows the files and directory structure
+      typical of a font project hosted on
+      the Google Fonts repo on GitHub ? """
+
+  # FIXME: Improve this with more details
+  #        about the expected structure.
+  return font.split("/")[-3] in ["ufl", "ofl", "apache"]
+
+
+@check(
+  id = 'com.google.fonts/check/repo/dirname_matches_nameid_1',
+  conditions = ['gfonts_repo_structure'],
+  misc_metadata = {
+    'request': 'https://github.com/googlefonts/fontbakery/issues/2302'
+  }
+)
+def com_google_fonts_check_repo_dirname_match_nameid_1(font,
+                                                       gfonts_repo_structure):
+  """Directory name in GFonts repo structure must match NameID 1."""
+  from fontbakery.utils import get_name_entry_strings
+  from fontTools.ttLib import TTFont
+
+  entry = get_name_entry_strings(TTFont(font), NameID.FONT_FAMILY_NAME)[0]
```

Comment:
This won't work if the first font is a non RIBBI font and it only has win name records. You may end up with Montserrat Light

Reply:
ok, got it!

I think that this should then be a family check instead. So that it receives `fonts`, selects the regular one, and reads the FONT FAMILY NAME record of that one.